### PR TITLE
Fix peers start conditions bug

### PR DESCRIPTION
**Pull request by OlivierHecart**
connect_peers returns false if not connected to avoid wrong start_condititions notif

### CODE REVIEW

**Review by oteffahi — APPROVED** [review]
Bug seems fixed.
LGTM.

File: zenoh/src/net/runtime/orchestrator.rs [hunk 1]
```diff
@@ -1024,7 +1024,7 @@ impl Runtime {
         }
     }
 
-    pub async fn connect_peer(&self, zid: &ZenohIdProto, locators: &[Locator]) {
+    pub async fn connect_peer(&self, zid: &ZenohIdProto, locators: &[Locator]) -> bool {
```
Comment (fuzzypixelz) [inline zenoh/src/net/runtime/orchestrator.rs:1027]: Correct me if I'm wrong. `Runtime::connect` returns true if it has opened a new Transport with `zid`. This seems to do the opposite.

Comment (oteffahi) [reply zenoh/src/net/runtime/orchestrator.rs:1027]: If I understand correctly, `connect_peer` only returns `false` if `Runtime::connect` returns `false`, and `Runtime::connect` returns `true` if it succeeds in opening a new transport with `zid`.
The only confusing bit to me is that if a connection is pending with `zid`, then both functions will return `false`.

Comment (fuzzypixelz) [reply zenoh/src/net/runtime/orchestrator.rs:1027]: Nevermind, it returns true if a Transport is opened with `zid`.

File: zenoh/src/net/runtime/orchestrator.rs [hunk 2]
```diff
@@ -911,7 +911,7 @@ impl Runtime {
         }
     }
 
-    /// Returns `true` if a new Transport instance has been opened with `zid`.
+    /// Returns `true` if a new Transport instance exists between us and `zid`.
```
Comment (oteffahi) [inline zenoh/src/net/runtime/orchestrator.rs:914]: This is misleading, as the `connect` function doesn't only check if a transport exists, but also tries to open one if it does not exist.

**Review by oteffahi — CHANGES_REQUESTED** [review]

File: zenoh/src/net/runtime/orchestrator.rs (hunk 2 above)
Comment (oteffahi) [reply zenoh/src/net/runtime/orchestrator.rs:914]:
```suggestion
    /// Opens a new transport with `zid`.
    /// Returns `true` if successful, or if a Transport instance already exists with `zid`.
```

**Review by oteffahi — APPROVED** [review]

**Pull request merged**